docs/wiki/Troubleshooting.md

```diff
@@ -0,0 +1,108 @@
+
+### `WARN Skip generating metric sample for broker because all broker metrics are missing`
+
+Sample log lines that accompany this warning might look like the following:
+```
+INFO Kicking off partition metric sampling for time range [1616448953094, 1616449073094], duration 120000 ms with timeout 120000 ms. (com.linkedin.kafka.cruisecontrol.monitor.sampling.MetricFetcherManager)
+INFO Finished sampling for topic partitions [__CruiseControlMetrics-0] in time range [1616448953094,1616449073094]. Collected 84318 metrics. (com.linkedin.kafka.cruisecontrol.monitor.sampling.CruiseControlMetricsReporterSampler)
+WARN Skip generating metric sample for broker 2 because all broker metrics are missing. (com.linkedin.kafka.cruisecontrol.monitor.sampling.SamplingUtils)
+WARN Skip generating metric sample for broker 0 because all broker metrics are missing. (com.linkedin.kafka.cruisecontrol.monitor.sampling.SamplingUtils)
+WARN Skip generating metric sample for broker 1 because all broker metrics are missing. (com.linkedin.kafka.cruisecontrol.monitor.sampling.SamplingUtils)
+INFO Generated 6333 partition metric samples and 3(3 skipped) broker metric samples for timestamp 1616449046081. (com.linkedin.kafka.cruisecontrol.monitor.sampling.CruiseControlMetricsProcessor)
+INFO Collected 6333 partition metric samples for 6333 partitions. Total partition assigned: 6333. (com.linkedin.kafka.cruisecontrol.monitor.sampling.SamplingFetcher)
+INFO Collected 3 broker metric samples for 3 brokers. (com.linkedin.kafka.cruisecontrol.monitor.sampling.SamplingFetcher)
+INFO Finished sampling in 654 ms. (com.linkedin.kafka.cruisecontrol.monitor.sampling.MetricFetcherManager)
+```
+
+This warning means that Cruise Control could not find broker metrics for the mentioned brokers (brokers `0`, `1`, `2`) in the `__CruiseControlMetrics` topic. It could, however, find broker metrics for 3 other brokers in the cluster. Without broker metrics, Cruise Control will not be able to generate a proper cluster model and might not be able to include the brokers with missing metrics in it poposal computations and task executions.
+
+### Resolution
+
+The general resolution is to ensure that the brokers with the missing metrics are able to produce metrics to the `__CruiseControlTopic`. Checking the following might help:
+
+1. Ensure that the Cruise Control metrics reporter jar is present on the brokers
+2. Ensure that you are using a version of Cruise Control compatible with the version of Kafka. For more information on version compatibility, see the [Environment Requirements](https://github.com/linkedin/cruise-control#environment-requirements) section
+
+## Contributing
+
+Found an error/interesting behaviour of Cruise Control and have a resolution for it? Consider adding that to this guide by following the [contributing guidelines](https://github.com/linkedin/cruise-control/blob/f8d272c4e3ea87b3ff7a66450957d60153af9e02/CONTRIBUTING.md) for the project.
```

Comment:
Nit: Should we use `https://github.com/linkedin/cruise-control/blob/master/CONTRIBUTING.md` rather than `https://github.com/linkedin/cruise-control/blob/f8d272c4e3ea87b3ff7a66450957d60153af9e02/CONTRIBUTING.md`?

Reply:
Thanks for pointing that out! Updated in 6094f90